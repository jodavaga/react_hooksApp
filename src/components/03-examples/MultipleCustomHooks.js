import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';

export const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter( 1 );
    const state = useFetch( `https://www.breakingbadapi.com/api/quotes/${ counter }` );


    // console.log(state);

    const { data, loading, error } = state;

    // destructured data object. Evaluate, if not null, return data[0]
    const { author, quote } = !!data && data[0];

    return (
        <div className="container">
            <h1>Breaking Bad Quotes</h1>
            <hr />
            {
                error && <div className="alert alert-danger"> Algo salio mal </div>
            }

            { loading 
                ? 
                    (
                        <div className="alert alert-info">Loading...</div>
                    )  
                :   
                    (
                        <div className="card">
                            <div className="card-body">
                                <blockquote className="blockquote text-right">
                                    <p>{ quote }</p>
                                    <footer className="blockquote-footer">{ author }</footer>
                                </blockquote>
                            </div>
                        </div>
                    )       
            }

            <button
                className="btn btn-primary mt-3"
                onClick={ () => increment() }
            >Next</button>

        </div>
    )
}