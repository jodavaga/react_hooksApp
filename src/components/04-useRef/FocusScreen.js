import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputElement = useRef(null);

    const handleFocus = () => {
        inputElement.current.select();
    }

    return (
        <div className="container">
            <h1>Focus useRef</h1>
            <hr />

            <input 
                type="text"
                className="form-control"
                placeholder="Full Name"
                ref={ inputElement }
            />

            <button
                className="btn btn-outline-primary mt-3"
                onClick={ handleFocus }
            >Focus</button>
        </div>
    )
}
