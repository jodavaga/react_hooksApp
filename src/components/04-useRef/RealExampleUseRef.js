import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleUseRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div className="container">
            <h1>RealExampleUseRef</h1>
            <hr />

            { show && <MultipleCustomHooks />}

            <button
                className="btn btn-outline-primary mt-3"
                onClick={ () => setShow( !show ) }
            >Show/Hide</button>
        </div>
    )
}
