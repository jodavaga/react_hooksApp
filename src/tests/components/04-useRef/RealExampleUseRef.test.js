import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleUseRef } from '../../../components/04-useRef/RealExampleUseRef';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';


describe('Test on <RealExampleUseRef />', () => {
    
    const wrapper = shallow( <RealExampleUseRef /> );
    
    test('should render component', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);
    });

    test('should display <MultipleCustomHooks />', () => {

        wrapper.find('button').simulate('click');

        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
    });
    
    
})
