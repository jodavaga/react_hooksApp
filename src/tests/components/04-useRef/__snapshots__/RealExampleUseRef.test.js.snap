// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <RealExampleUseRef /> should render component 1`] = `
<div
  className="container"
>
  <h1>
    RealExampleUseRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
